### PR TITLE
jewel: core: test_envlibrados_for_rocksdb.yaml fails on crypto restart

### PR DESCRIPTION
**Pull request by smithfarm**
http://tracker.ceph.com/issues/20460

### CODE REVIEW

**Comment by smithfarm** [comment]
test this please

**Comment by smithfarm** [comment]
Marking DNM until we can get a clean "make check" and figure out why wip-jewel-backports only builds for Xenial.

**Comment by smithfarm** [comment]
Jenkins re-test this please

**Comment by smithfarm** [comment]
@tchaikov This causes a massive failure in rados and upgrade/client-upgrade suites. Any idea why?

http://pulpito.ceph.com/smithfarm-2017-09-01_17:10:21-upgrade:client-upgrade-wip-20460-jewel-distro-basic-vps/

**Comment by tchaikov** [comment]
@smithfarm will take a look next Monday.

**Comment by tchaikov** [comment]
i am trying to reproduce it locally.

File: src/osdc/Objecter.h [hunk 1]
```diff
@@ -2058,6 +2058,9 @@ class Objecter : public md_config_obs_t, public Dispatcher {
   }
   void ms_fast_dispatch(Message *m) {
     ms_dispatch(m);
+    if (!ms_dispatch(m)) {
```
Comment (tchaikov) [inline src/osdc/Objecter.h]: @smithfarm this backport is buggy.

we should call `ms_dispatch()` for only one time. so this patch should instead look like
```diff
diff --git a/src/osdc/Objecter.h b/src/osdc/Objecter.h
index 8167306dd8..2955a071e7 100644
--- a/src/osdc/Objecter.h
+++ b/src/osdc/Objecter.h
@@ -2057,7 +2057,9 @@ private:
     }
   }
   void ms_fast_dispatch(Message *m) {
-    ms_dispatch(m);
+    if (!ms_dispatch(m)) {
+      m->put();
+    }
   }

   void handle_osd_op_reply(class MOSDOpReply *m);
```

Comment (smithfarm) [reply src/osdc/Objecter.h]: @tchaikov ACK I overlooked that line when resolving the cherry-pick conflict. Fixed now. Thanks a lot!

**Comment by smithfarm** [comment]
jenkins build docs

**Comment by smithfarm** [comment]
jenkins test docs

**Comment by smithfarm** [comment]
Passed a rados suite at http://tracker.ceph.com/issues/20613#note-48 with some failures that are, I believe, unrelated.

**Pull request merged**